aten/src/THC/generic/THCTensorCopy.cpp

```diff
@@ -94,9 +96,11 @@ void THTensor_(copyCuda)(THCState *state, THTensor *self, struct THCTensor *src)
       THTensor *srcf = THTensor_(newWithSize)(src->sizes(), {});          \
                                                                           \
       THTensor_(copyCuda)(state, srcf, src);                              \
-      TH_CONCAT_4(TH,TYPEC,Tensor_copy,Real)(self, srcf);                 \
+      at::Tensor srcf_wrap = THTensor_wrap(srcf);                         \
+      at::Tensor self_wrap = THTensor_wrap(self);                         \
+      at::_copy_(self_wrap, srcf_wrap);                                    \
                                                                           \
-      c10::raw::intrusive_ptr::decref(srcf);                                              \
+      c10::raw::intrusive_ptr::decref(srcf);                              \
```

Comment:
nit: Since `srcf` is never used after here, you don't need to incref/decref; you can just steal it entirely, saving an incref/decref. This is unlikely to show up in a singlethreaded benchmark, but we should practice good hygiene.

Actually, going one step further, `srcf` is allocated using `THTensor_(newWithSize)`, but you could use an equivalent ATen function `at::empty(src->sizes())` to have the same effect, and get a `at::Tensor` while you're at it.